### PR TITLE
update pvDatabaseJava.html using new doc conventions

### PR DESCRIPTION
**Pull request by mrkraimer**

### CODE REVIEW

**Comment by ralphlange** [comment]
@mrkraimer:
It is not necessary to merge the target branch into your pull requests.
This creates extra commits that are confusing.

Automated builds and tests (CloudBees PR-builder jobs) are done on the result of the proposed merge.

File: documentation/pvDatabaseJava.html [hunk 1]
```diff
@@ -36,29 +36,8 @@
 
 <div class="head">
 <h1>pvDatabaseJava</h1>
-<!-- Maturity: Working Draft or Request for Comments, or Recommendation, and date. -->
+<h2 class="nocount">Release 4.2 - 2016.03.21</h2>
```
Comment (ralphlange) [inline documentation/pvDatabaseJava.html:39]: @dhickin proposed to use the ...-DEV notation here, to clarify that this is documentation in progress,

**Comment by mrkraimer** [comment]
On 03/21/2016 12:08 PM, Ralph Lange wrote:

> @mrkraimer https://github.com/mrkraimer:
> It is not necessary to merge the target branch into your pull requests.
> This creates extra commits that are confusing.

But if I do not merge the target branch then github shows that my 
repository is behind the target.
Maybe I did the merge of  the target branch before the cloudbees builds 
were complete?
In the future I will wait at least a day after the merge is complete.

**Comment by ralphlange** [comment]
On 22/03/2016 12:22, Marty Kraimer wrote:

> On 03/21/2016 12:08 PM, Ralph Lange wrote:
> 
> > It is not necessary to merge the target branch into your pull requests.
> > This creates extra commits that are confusing.
> 
> But if I do not merge the target branch then github shows that my
> repository is behind the target.
> Maybe I did the merge of the target branch before the cloudbees builds
> were complete?
> In the future I will wait at least a day after the merge is complete.

Yes, your repo is behind the target. That's because other changes were 
pushed to the target that your clone does not have.
So what?
GitHub shows you that "This branch has no conflicts with the base 
branch". That means it can be merged straight away.
There is absolutely no need to rebase your branch to the current state 
of the target before merging.

> > Automated builds and tests (CloudBees PR-builder jobs) are done on 
> > the result of the proposed merge.

The PR-builder has already built and tested the _result_ of 
merging your proposal into the target branch.
This is a service so that you don't have to do the test merge and build 
locally.

Only in the case of conflicts, which are clearly shown in the "checks" 
box of your PR, merging the target branch can be useful as it might 
allow you to solve the expected conflicts on your PR branch. Once you 
have done that, the checks will turn green again.

The CloudBees PR-builder jobs are just a way of testing if the result of 
the proposed merge can be built and tests ok. They are not required for 
the merge, GitHub is not aware of the CloudBees jobs at all. (It's the 
CloudBees jobs that are writing comments and add build results to the PR 
on GitHub.)

Bottom line: If the PR has no conflicts with the target, please don't 
merge the target back into the source.
If you have a good reason for checking the merge result locally 
beforehand, clone the target branch into a local temporary branch, and 
merge your PR into it. The result is exactly what the target master 
would look like with your PR merged. You can build and test away on that 
temporary branch, and if everything's good, merge your original PR into 
target master without the confusing back-merge.

To support that, it is good practice to not develop on the master branch 
of your clone, but on a dedicated development branch, and use that as 
the source of the PR. This allows you to often update you clone's master 
from the original source (i.e. target master) without interfering with 
your developments.

**Pull request merged**